tests/test_plugins/test_error_responses.py

```diff
@@ -0,0 +1,63 @@
+from typing import Type
+
+from starlette import status
+from starlette.applications import Starlette
+from starlette.middleware import Middleware
+from starlette.responses import JSONResponse, Response
+from starlette.testclient import TestClient
+
+from starlette_context import plugins
+from starlette_context.header_keys import HeaderKeys
+from starlette_context.middleware import (
+    ContextMiddleware,
+    RawContextMiddleware,
+)
+
+
+def gen_middleware_config(
+    middleware_class: Type, response: Response
+) -> TestClient:
+    middleware = [
+        Middleware(
+            middleware_class,
+            plugins=(plugins.RequestIdPlugin(error_response=response),),
+        )
+    ]
+    app = Starlette(middleware=middleware)
+    client = TestClient(app)
+    return client
+
+
+def test_invalid_request_id_returns_specified_response_raw_middleware():
+    content = {"Error": "Invalid X-Request-ID"}
+    response = JSONResponse(
+        content=content, status_code=status.HTTP_422_UNPROCESSABLE_ENTITY
```

Comment:
I am also thinking it should be either 422 or 400. I'm not sure as it's a header, so kind of a request metadata. What's your reasoning?

Reply:
default is a 400, which I think is a fair default, and this tests that user can change it to whatever they want if they disagree.

Reply:
as for the reasoning, I would personally prefer to assign 422 to body validation as they would represent objects, which would be close enough to the 422 "entity". 400 is the generic client-side so this fits better as a header-related validation default imo. But this is obviously something that people should be left free to choose.